cmd/crank/update.go

```diff
@@ -0,0 +1,151 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/alecthomas/kong"
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	v1 "github.com/crossplane/crossplane/apis/pkg/v1"
+	typedclient "github.com/crossplane/crossplane/internal/client/clientset/versioned/typed/pkg/v1"
+	"github.com/crossplane/crossplane/internal/xpkg"
+
+	// Load all the auth plugins for the cloud providers.
+	_ "k8s.io/client-go/plugin/pkg/client/auth"
+)
+
+// updateCmd updates a package.
+type updateCmd struct {
+	Configuration updateConfigCmd   `cmd:"" help:"Update a Configuration package."`
+	Provider      updateProviderCmd `cmd:"" help:"Update a Provider package."`
+}
+
+// Run runs the update cmd.
+func (c *updateCmd) Run(b *buildChild) error {
+	return nil
+}
+
+// updateConfigCmd updates a Configuration.
+type updateConfigCmd struct {
+	Package string `arg:"" help:"Image containing Configuration package."`
+	Tag     string `arg:"" help:"Updated tag for Configuration package."`
+
+	Name                 string   `arg:"" optional:"" help:"Name of Configuration."`
+	RevisionHistoryLimit int64    `short:"rl" help:"Revision history limit."`
+	ManualActivation     bool     `short:"m" help:"Enable manual revision activation policy."`
+	PackagePullSecrets   []string `help:"List of secrets used to pull package."`
+}
+
+// Run runs the Configuration update cmd.
+func (c *updateConfigCmd) Run(k *kong.Context) error {
+	rap := v1.AutomaticActivation
+	if c.ManualActivation {
+		rap = v1.ManualActivation
+	}
+	pkgName := c.Name
+	if pkgName == "" {
+		ref, err := name.ParseReference(c.Package)
+		if err != nil {
+			return errors.Wrap(err, errPkgIdentifier)
+		}
+		pkgName = xpkg.ToDNSLabel(ref.Context().RepositoryStr())
+	}
+	packagePullSecrets := make([]corev1.LocalObjectReference, len(c.PackagePullSecrets))
+	for i, s := range c.PackagePullSecrets {
+		packagePullSecrets[i] = corev1.LocalObjectReference{
+			Name: s,
+		}
+	}
+	pkgReference, _ := name.ParseReference(c.Package)
+	newPkg := ""
+	if strings.HasPrefix(c.Tag, "sha256") {
+		newPkg = pkgReference.Context().Digest(c.Tag).Name()
+	} else {
+		newPkg = pkgReference.Context().Tag(c.Tag).Name()
+	}
+	cr := &v1.Configuration{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: pkgName,
+		},
+		Spec: v1.ConfigurationSpec{
+			PackageSpec: v1.PackageSpec{
+				Package:                  newPkg,
+				RevisionActivationPolicy: &rap,
+				RevisionHistoryLimit:     &c.RevisionHistoryLimit,
+				PackagePullSecrets:       packagePullSecrets,
```

Comment:
This will likely overwrite these fields to their zero values if a user does not provide values for the flags. We should only update if the user provides args, which could be identified by using references and checking for nil